File: png2mat.m
imgPath = '/Users/xxx/Documents/yyy/';        % 图像库路径
imgDir  = dir([imgPath '*.png']); % 遍历所有jpg格式文件

for i = 1:length(imgDir)          % 遍历结构体就可以一一处理图片了
    img = imread([imgPath imgDir(i).name]); %读取每张图片
    gray = rgb2gray(img);
    temp3 = double(gray) / 255;
    a = ['/Users/xxx/Documents/yyy/(', int2str(i), ').mat'];
    save(a, 'temp3');
    fprintf('done\n');
end

File: png2mat.py
import glob
import numpy as np
import scipy.io as io
import torchvision
from PIL import Image

src_dir = '/Users/xxx/Documents/数据集/URBAN100_PNG/'
save_dir = '/Users/xxx/Documents/数据集/URBAN100_MAT'
file_list = glob.glob(src_dir+'*.png')
print('Start...')

for i in range(file_list.__len__()):
    img = Image.open(file_list[i])

    img = torchvision.transforms.ToTensor()(img)
    img = torchvision.transforms.Grayscale(num_output_channels=1)(img).squeeze()
    img = img.numpy()
    io.savemat(save_dir + '/({}).mat'.format(i + 1), {'temp3': img})
print('done')
